### DISCUSSION TITLE
Error in if (nrow(d) == 0) return(NULL) : argument is of length zero

### DISCUSSION THREAD

**Discussion by ikram2779** [MRCIEU/TwoSampleMR#491] [Q&A]
>extract_instruments("ukb-b-7992", access_token = NULL)
Error in if (nrow(d) == 0) return(NULL) : argument is of length zero

please help

**Comment by PPKstar** [comment]
I meet the problem too. And when I use "extract_outcome_data", it just return NULL.

**Reply by PPKstar** [reply]
I have tried to update "TwoSampleMR" package, but it didn't work.

**Reply by zhengjinjinjin** [reply]
Me too!

**Reply by Ali-mhm** [reply]
> I meet the problem too. And when I use "extract_outcome_data", it just return NULL.


in the TWOsampleMR package whenever you receive a weird error, it is probably a server issue such as this error, or when you are doing clump, that uses this line of code:
```
options(ieugwasr_api = 'gwas-api.mrcieu.ac.uk/')
```

**Comment by Gseven222** [comment]
I meet the problem too.

**Comment by DC-Jer** [comment]
Confront with same problem from yesterday afternoon when using the [API](https://gwas-api.mrcieu.ac.uk/) options to query the data.
(https://github.com/MRCIEU/TwoSampleMR/assets/37207740/133fc0fb-4cb4-4913-8685-e10d7ca9586b)

**Comment by fcf01** [comment]
I meet this issue too, and still undo now, anybody could help me 
expo_rt<-extract_instruments(outcome= "balabalabla,p1 = 1e-5,
+                                                            clump = T,
+                                                            p2 = 1e-5,
+                                                            r2 = 0.001,
+                                                            kb = 10000) 
and tell me : Error in if (nrow(d) == 0) return(NULL) : argument is of length zero

**Comment by zhuhhfsdfsd** [comment]
I meet the problem too.

**Comment by wuyisky** [comment]
I meet the problem too.

**Comment by cykky** [comment]
something wrong with the server?

**Reply by Spasmodic114** [reply]
Yesterday I accessed the website (https://gwas.mrcieu.ac.uk) and it showed Server Error 500. Now I can access the website but I met the same problem with TwoSampleMR package. :(

**Reply by xiaobaiyaobianqiang** [reply]
I meet the problem too.

**Comment by Castiel4517** [comment]
Me too, is that a bug?

**Reply by xiaobaiyaobianqiang** [reply]
how to solve

**Comment by ikram2779** [comment]
it is working today as usual. probably it was server issue

**Comment by Spasmodic114** [comment]
I met the problem again 1 hour ago :(
I wonder if it is the server error or other problems?

**Reply by Spasmodic114** [reply]
It showed "server code: 502" before: "Error in if (nrow(d) == 0) return(NULL) : argument is of length zero"

**Comment by xiaobaiyaobianqiang** [comment]
I meet the problem too.how to solve

**Comment by crebli** [comment]
me too even I got the access token.

**Comment by xiaobaiyaobianqiang** [comment]
I solve the question by
options(ieugwasr_api = 'gwas-api.mrcieu.ac.uk/')

**Reply by Spasmodic114** [reply]
I met this problem even I got the token :(

**Reply by zhangjl520** [reply]
Thank u, i solved the problem in this way.

**Reply by cmulph** [reply]
Thank U

**Reply by Fin-beep** [reply]
Hello, what are the specific steps

**Reply by Lecter-HHH** [reply]
Thank you! This method works!

**Reply by SUMCZY2021** [reply]
this method not work for me. i still get "Error in if (nrow(d) == 0) return(NULL) : argument is of length zero". does anyone have other solution?

**Reply by SUMCZY2021** [reply]
che<-extract_instruments(outcomes='ukb-b-1489', clump=TRUE)
Error in if (nrow(d) == 0) return(NULL) : argument is of length zero".

**Comment by crebli** [comment]
Nice! Thank you!发自我的荣耀手机-------- 原始邮件 --------发件人： xiaobaiyaobianqiang ***@***.***>日期： 2024年5月5日周日 下午4:52收件人： MRCIEU/TwoSampleMR ***@***.***>抄送： crebli ***@***.***>, Comment ***@***.***>主    题： Re: [MRCIEU/TwoSampleMR] Error in if (nrow(d) == 0) return(NULL) : argument is of length zero (Discussion #491)
I solve the question by
options(ieugwasr_api = 'gwas-api.mrcieu.ac.uk/')

**Reply by vividchan** [reply]
I also solved this question by your method, thanks

**Reply by yuwei0529** [reply]
thank you very much,i solved too

**Reply by Fin-beep** [reply]
Hello, what are the specific steps

**Comment by a1141539455** [comment]
I met this problem even I got the token：
options(ieugwasr_api = 'gwas-api.mrcieu.ac.uk/')

**Reply by vividchan** [reply]
You can try to use another dataset to validate whether your previous dataset is not readable. Because I find out that some datasets are not available.

**Reply by a1141539455** [reply]
I have tried. I was able to extract the instrumental variable data set before, but now I keep getting this error. :(
Can you use it normally now?

**Reply by vividchan** [reply]
Yes, the code works under my R enviroment. Maybe you can try to re-download the ieugwasr package.

**Reply by a1141539455** [reply]
I have re-installed ieugwasr1.0.0 and twosamplemr0.6.2 but it still gives the error :(

**Comment by ManqiZh** [comment]
The same problem is repeated today.

**Reply by MedTiLab** [reply]
how to slove it , it repeat all days!

**Comment by AliiReezaa** [comment]
options(ieugwasr_api = 'gwas-api.mrcieu.ac.uk/') This method was working for me for a long time; unfortunately, starting yesterday, it is no longer working. Do you have any idea how to resolve this issue?

**Reply by remlapmot** [reply]
We have told users many times to please stop using the old API address - there is no need to set this in `options()` anymore.

**Reply by AliiReezaa** [reply]
So how to do it if we are not supposed to use this API address? What is the new one?

**Reply by remlapmot** [reply]
With the latest versions of TwoSampleMR and ieugwasr it is already set for you - just follow the authentication steps to setup a token - https://mrcieu.github.io/ieugwasr/articles/guide.html#authentication .

**Reply by Ali-mhm** [reply]
> With the latest versions of TwoSampleMR and ieugwasr it is already set for you - just follow the authentication steps to setup a token - https://mrcieu.github.io/ieugwasr/articles/guide.html#authentication .

You are right but
For two weeks, whenever you try to obtain a token and sign in, it results in an overload error.

**Reply by ldbldb** [reply]
> 我们已经多次告诉用户，请停止使用旧的 API 地址 - 无需再设置它。`options()`



> 使用最新版本的 TwoSampleMR 和 ieugwasr，它已经为您设置好了 - 只需按照身份验证步骤设置令牌 - https://mrcieu.github.io/ieugwasr/articles/guide.html#authentication 。

Hello, using the latest version of TwoSampleMR and ieugwasr, I still get the error: error in if (out == “[]”) {: parameter length is zero, then I can get the result after changing the parameter to: clump = FALSE, may I ask what is the reason?

**Comment by AliiReezaa** [comment]
I appreciate your informative guide. It finally worked.
best regards,
Ali Reza

**Comment by fe1m22** [comment]
I have been having difficulties connecting to the API using extract_instruments as part of TwoSampleMR package. I have tried new tokens, re-installing the ieugwasr and TwoSampleMR packages and updating R studio. Any advice would be very helpful. I get the following error:

Error in if (out == "[]") { : argument is of length zero

**Reply by fe1m22** [reply]
I have just read the above comment and clump=F seems to work but then the SNPs are not clumped. Does anyone know a work around? Any help would be appreciated.

**Reply by ldbldb** [reply]
After modifying clump=F, in the next step just clump the exposed SNPs  and merge them with the outcome SNPs , but I'm not sure if this is the right way to go about it

**Comment by HuXuantao** [comment]
I'm pretty sure there is sth wrong with the server or the package code. I tried to debug for a whole day and here is the findings:
1. When using default args, you can get the normal returns from functions like `extract_outcome` etc. (I compared a new returns with my previous one produced by a successful run)
2. BUT if you change the default setup of args (for me, i tried to merely change `r2 = 0.001` to `r2 = 0.0011`), you will get a "`(nrow(d) == 0) return(NULL)`". (I also compared a run using the same args with one of my previous successful run. it just failed.)

I have already updated twosamplemr (0.6.11) and ieugwasr (1.0.1) package and got my IEU token working. But these seems having nothing to do with the ERROR. lol! 
Frustrating...

**Reply by HuXuantao** [reply]
Today, it works...
